crates/forge/src/running.rs

```diff
@@ -229,35 +234,42 @@ pub(crate) fn run_test_case(
         &string_to_hint,
     );
 
-    Ok(runner.run_function(
+    match runner.run_function(
         runner.find_function(case.name.as_str())?,
         &mut test_execution_syscall_handler,
         hints_dict,
         instructions,
         builtins,
-    )?)
+    ) {
+        Ok(result) => Ok(TestCaseSummary::from_run_result(result, case, args)),
+        // CairoRunError comes from VirtualMachineError which may come from HintException that originates in the TestSyscallExecutionHandler
+        Err(RunnerError::CairoRunError(error)) => Ok(TestCaseSummary::Failed {
+            name: case.name.clone(),
+            msg: Some(format!(
+                "\n    {}\n",
+                error.to_string().replace(" Custom Hint Error: ", "\n    ")
+            )),
+            arguments: args,
+            fuzzing_statistic: None,
+        }),
+        Err(err) => Err(err.into()),
+    }
 }
 
 fn extract_test_case_summary(
```

Comment:
That's a weird flow to me. `TestCaseSummary` is already extracted in this function argument. We should have this handling in one place, there's no benefit to wrapping the error in two different places.

Reply:
The reason for that change is to be able to add  additional info to `TestCaseSummary`, e.g. number of the block used (that's why I did this PR, see description), if we got to `run_function` call . It is a compromise between doing weird `match` statements in every place `?` is used in `run_test_case` and passing additional info to `TestCrateSummary`

Reply:
I don't understand why cannot the arms of match in `run_test_case` and in `extract_test_case_summary` be combined. I'd rather have a one large match handling all errors than artificially splitting it into two functions.

Reply:
> I don't think running.rs should know anything about TestCaseSummary (stuff like fuzzing statistics and block number) it should be handled in an upper layer (cli layer). We should differentiate between TestCaseResult and summary and don't leak this stuff here as this feeds the complexity.

@MaksymilianDemitraszek out of scope of this PR, should be done with #839